From d24bb9c4b5d3b0bb2bd5dd922bae3fce894ab87e Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Tue, 1 Oct 2019 10:19:50 +0100
Subject: [PATCH] overlays: Remove hack from uart0 overlay

The uart0 overlay contained a hack to return GPIOs 14 and 15 to inputs
when the UART0 function was moved to alternative pins. This has the
unwanted side effect of claiming GPIOs 14 & 15, preventing them being
used for something else.

See: https://github.com/raspberrypi/linux/issues/2856
     https://www.raspberrypi.org/forums/viewtopic.php?f=98&t=252911

Signed-off-by: Stefan Enge <stefan.enge@escatec.com>
Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/overlays/uart0-overlay.dts | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

--- a/arch/arm/boot/dts/overlays/uart0-overlay.dts
+++ b/arch/arm/boot/dts/overlays/uart0-overlay.dts
@@ -17,17 +17,16 @@
 		target = <&gpio>;
 		__overlay__ {
 			uart0_pins: uart0_pins {
-				brcm,pins = <14 15 14 15>;
-				brcm,function = <0 0 4 4>; /* alt0 */
-				brcm,pull = <0 0 0 2>;
+				brcm,pins = <14 15>;
+				brcm,function = <4>; /* alt0 */
+				brcm,pull = <0 2>;
 			};
 		};
 	};
 
 	__overrides__ {
-		txd0_pin = <&uart0_pins>,"brcm,pins:8";
-		rxd0_pin = <&uart0_pins>,"brcm,pins:12";
-		pin_func = <&uart0_pins>,"brcm,function:8",
-			   <&uart0_pins>,"brcm,function:12";
+		txd0_pin = <&uart0_pins>,"brcm,pins:0";
+		rxd0_pin = <&uart0_pins>,"brcm,pins:4";
+		pin_func = <&uart0_pins>,"brcm,function:0";
 	};
 };
